### PR TITLE
Update sqlparser-derive to use `syn 2.0`

### PR DESCRIPTION
**Pull request by serprex**
https://github.com/dtolnay/syn/releases/tag/2.0.0

### CODE REVIEW

**Comment by tobyhede** [comment]
@serprex Some more context on this change would be useful ... I assume it is updating syn?

**Comment by serprex** [comment]
Yes

File: derive/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,6 @@ edition = "2021"
 proc-macro = true
 
 [dependencies]
-syn = "1.0"
+syn = { version = "2.0", features = ["full"] }
```
Comment (alamb) [inline derive/Cargo.toml:21]: Do we really need "features=full"? Keeping down the dependency list for this crate is important I think given how low level it is

**Comment by alamb** [comment]
Thank you for this contribution @serprex

File: derive/Cargo.toml (hunk 1 above)
Comment (serprex) [reply derive/Cargo.toml:21]: > full — Data structures for representing the syntax tree of all valid Rust source code, including items and expressions.

Code did not compile without enabling full feature

**Comment by serprex** [comment]
Won't have time this weekend to look into, but will look at how to run tests & fix next week (I only ran `cargo test` in derive crate)

**Comment by alamb** [comment]
Thanks @serprex .  Converting to a draft while review feedback is being addressed. Please mark it ready for review when it is ready for another look

File: derive/Cargo.toml [hunk 2]
```diff
@@ -18,6 +18,6 @@ edition = "2021"
 proc-macro = true
 
 [dependencies]
-syn = "1.0"
+syn = { version = "2.0", default-features = false, features = ["printing", "parsing", "derive", "proc-macro"] }
```
Comment (alamb) [inline derive/Cargo.toml:21]: It seems according to 
https://docs.rs/syn/latest/syn/index.html#optional-features

Listing these 4 will at least avoid

> clone-impls (enabled by default) — Clone impls for all syntax tree types.

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @serprex 

I took the liberty of merging up from main and reducing the feature set used

**Comment by alamb** [comment]
Thanks again @serprex  and @tobyhede

**Pull request merged**